modules/base/include/modules/base/processors/filecache.h

```diff
@@ -138,6 +105,138 @@ void updateFilenameFilters(const DataReaderFactory& rf, const DataWriterFactory&
 
 }  // namespace detail
 
+class IVW_MODULE_BASE_API CacheBase : public Processor, public ProcessorNetworkEvaluationObserver {
+public:
+    explicit CacheBase(InviwoApplication* app);
+    virtual bool isConnectionActive(Inport* inport, Outport* outport) const override;
+    virtual void onProcessorNetworkEvaluationBegin() override;
+
+    virtual bool hasCache(std::string_view key) = 0;
+
+protected:
+    void writeXML() const;
+
+    BoolProperty enabled_;
+    DirectoryProperty cacheDir_;
+    DirectoryProperty refDir_;
+    StringProperty currentKey_;
+
+    bool isCached_ = false;
+    std::string key_;
+    std::pmr::string xml_;
+};
+
+template <typename DataType>
+struct ReaderWriter {
+    explicit ReaderWriter(InviwoApplication* app)
+        : extensions{"readerWriter", "Data Reader And Writer"}
```

Comment:
Should the text not say "File format" or "File extension"?

Reply:
What is relevant here is what reader/writer that gets used. not what the acctual file ending is, so no

Reply:
I see. I would simply use "Reader/Writer" in that case. A bit less verbose. Does not matter much either